src/Controllers/UserController.js

```diff
@@ -191,36 +197,43 @@ export class UserController extends AdaptableController {
     );
   }
 
-  sendPasswordResetEmail(email) {
+  async sendPasswordResetEmail(email) {
     if (!this.adapter) {
       throw 'Trying to send a reset password but no adapter is set';
       //  TODO: No adapter?
     }
-
-    return this.setPasswordResetToken(email).then(user => {
-      const token = encodeURIComponent(user._perishable_token);
-      const username = encodeURIComponent(user.username);
-
-      const link = buildEmailLink(
-        this.config.requestResetPasswordURL,
-        username,
-        token,
-        this.config
-      );
-      const options = {
-        appName: this.config.appName,
-        link: link,
-        user: inflate('_User', user),
-      };
-
-      if (this.adapter.sendPasswordResetEmail) {
-        this.adapter.sendPasswordResetEmail(options);
-      } else {
-        this.adapter.sendMail(this.defaultResetPasswordEmail(options));
+    let user;
+    if (
+      this.config.passwordPolicy &&
+      this.config.passwordPolicy.resetTokenReuseIfValid &&
+      this.config.passwordPolicy.resetTokenValidityDuration
+    ) {
+      try {
+        user = await this.checkResetTokenValidity(email);
+      } catch (e) {
+        /* */
```

Comment:
Could you explain?

Reply:
If `passwordPolicy.resetTokenReuseIfValid` and passwordPolicy.resetTokenValidityDuration, check if the token is valid. If the token is valid, user will be requested user, otherwise the checkResetTokenValidity will throw an error (so user will be null), and a new token will need to be generated

Reply:
Maybe it would be better to split `checkResetTokenValidity` in two functions so we could call the verification from here and just have true/false instead of a try catch (with no catch). I'm afraid another error can happen here and the developer will not even know.